lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -6326,11 +6322,20 @@ LogicalResult LayerBlockOp::verify() {
           if (isa<RefDefineOp>(connect))
             return WalkResult::advance();
 
-          // We can drive any destination inside the current layerblock.
+          // Verify that connects only drive values declared in the layer block.
+          // If we see a non-passive connect destination, then verify that the
+          // source is in the same layer block so that the source is not driven.
```

Comment:
I think the PR is just wrong. I had started to write his using `foldFlow` which I think is how it has to work if the operation is a `connect`. The check for `matchingconnect` and simpler operations is easier. Let me dig into this.

Reply:
`foldFlow` isn't quite right either. In order for this to work correctly, it needs to know if something is a source, sink, or duplex. However, this isn't the usual `foldFlow` computation which looks up towards the declaration, but looks down towards the leaves of the type.

It's annoying because I think it would be just a passive check if type canonicalization was possible.

Reply:
This is exceedingly annoying to verify... It requires examining the type in detail. In this PR, I opted to _not_ verify this. I don't like this, however, non-passive types are incredibly short-lived in the compiler. These are almost immediately removed due to everything becoming `matchingconnect`. This saves some tedious, expensive verification for a code path that is pretty rare.

I put a massive TODO in the code explaining this and will expect to revisit this in the future.

I'd like to merge this as-is and we can tighten this up later.

Reply:
Sounds good!